File: LAB 1/MATLAB/MANDELBROT.m
clc;
clear;
close all;

max_iter = 200;

x_vals = linspace(-2.5, 1, 1000);
y_vals = linspace(-1.5, 1.5, 1000);
[x_grid, y_grid] = meshgrid(x_vals, y_vals);
c_grid = x_grid + 1i*y_grid;

z_grid = zeros(size(c_grid));
mandelbrot = zeros(size(c_grid));

for k = 1:max_iter
    z_grid = z_grid.^2 + c_grid;
    mask = abs(z_grid) <= 2;
    mandelbrot(mask) = mandelbrot(mask) + 1;
end

figure;
imagesc(x_vals, y_vals, mandelbrot);
axis equal tight;
colormap(hot);
colorbar;
title('Mandelbrot Set');
xlabel('Real Axis');
ylabel('Imaginary Axis');

File: LAB 1/Python/MANDELBROT.py
import numpy as np
import matplotlib.pyplot as plt

max_iter = 200

x_vals = np.linspace(-2.5, 1, 1000)
y_vals = np.linspace(-1.5, 1.5, 1000)
x_grid, y_grid = np.meshgrid(x_vals, y_vals)
c_grid = x_grid + 1j * y_grid

z_grid = np.zeros_like(c_grid, dtype=complex)
mandelbrot = np.zeros(c_grid.shape)

for _ in range(max_iter):
    z_grid = z_grid**2 + c_grid
    mask = np.abs(z_grid) <= 2
    mandelbrot[mask] += 1

plt.figure()
plt.imshow(
    mandelbrot,
    extent=[x_vals.min(), x_vals.max(), y_vals.min(), y_vals.max()],
    cmap='hot',
    origin='lower'
)
plt.colorbar()
plt.title("Mandelbrot Set")
plt.xlabel("Real Axis")
plt.ylabel("Imaginary Axis")
plt.axis("equal")
plt.tight_layout()
plt.show()
